### ISSUE TITLE
Incorrect fragment generation for same functions in the translation unit.

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-prepo#40]
Compile the following code:
```
void t1()
{
  int a = 10;
  a++;
}

void t2()
{
  int b = 10;
  b++;
}
```
using
```
$ rm clang.db
$ clang -O0 -S -emit-llvm -fno-exceptions -target x86_64-pc-linux-gnu-repo -o repo.ll test.c
$ llc -filetype=obj -o repo.o repo.ll
$ pstore-dump –all-fragments clang.db

```
The list of fragments in the repository is:

```
  fragments :
      - digest   : 649cd21c6e3829c7a776f287c978c731
        fragment :
            - type     : text
              contents :
                  align   : 16
                  data    :
                      - 0:      pushq   %rbp
                      - 1:      movq    %rsp, %rbp
                      - 4:      movl    $10, -4(%rbp)
                      - 11:     movl    -4(%rbp), %eax
                      - 14:     addl    $1, %eax
                      - 17:     movl    %eax, -4(%rbp)
                      - 20:     popq    %rbp
                      - 21:     retq
                      - 22:     nop
                      - 23:     nop
                      - 24:     nop
                      - 25:     nop
                      - 26:     nop
                      - 27:     nop
                      - 28:     nop
                      - 29:     nop
                      - 30:     nop
                      - 31:     nop
                      - 32:     pushq   %rbp
                      - 33:     movq    %rsp, %rbp
                      - 36:     movl    $10, -4(%rbp)
                      - 43:     movl    -4(%rbp), %eax
                      - 46:     addl    $1, %eax
                      - 49:     movl    %eax, -4(%rbp)
                      - 52:     popq    %rbp
                      - 53:     retq
                  ifixups : [ ]
                  xfixups : [ ]
```
The function t1 and t2 have been merge into a single fragment, whose data is wrong.

**Comment by MaggieYingYi** [comment]
Fixed in commit 47cd545.

**Comment by MaggieYingYi** [comment]
A new issue is caused by the previous commit.

The definition of a global may be discarded if it is not used in its compilation unit. If a global has been pruned and referred to a discarded global in its compilation unit, a missing fragment error is met during the code generation stage.
